opensearch/sycamore-opensearch.sh

```diff
@@ -427,4 +433,61 @@ setup_transient() {
     deploy_model "${OPENAI_MODEL_ID}" "${OPENAI_TASK_ID}" "OpenAI"
 }
 
+create_certificates() {
+    local HOST="${SSL_HOSTNAME:-localhost}"
+    local DAYS=10000
+
+    pushd config
+    local OLDMASK=$(umask -p)
+    umask 077
+
+    # 1. Make fake certificate authority (CA) certificate.  OpenSearch
+    # requires a root certificate to be specified.
+    openssl req -batch -x509 -newkey rsa:4096 -days "${DAYS}" \
+    -subj "/C=US/ST=California/O=Aryn.ai/CN=Fake CA" \
+    -extensions v3_ca -noenc -keyout cakey.pem -out cacert.pem 2> /dev/null
+    echo "Created CA certificate"
+
+    # 2a. Create certificate signing request (CSR) for the node certificate.
+    openssl req -batch -newkey rsa:4096 \
+    -subj "/C=US/ST=California/O=Aryn.ai/CN=${HOST}" \
+    -extensions v3_req -addext "basicConstraints=critical,CA:FALSE" \
+    -addext "subjectAltName=DNS:${HOST}" \
+    -noenc -keyout node-key.pem -out node-req.pem 2> /dev/null
+
+    # 2b. Use the fake CA to sign the node CSR, yielding a certificate.
+    openssl x509 -req -CA cacert.pem -CAkey cakey.pem \
+    -copy_extensions copy -days "${DAYS}" \
+    -in node-req.pem -out node-cert.pem 2> /dev/null
+    echo "Created node certificate"
+
+    # 3a. Create certificate signing request (CSR) for the admin certificate.
+    openssl req -batch -newkey rsa:4096 \
+    -subj "/C=US/ST=California/O=Aryn.ai/CN=Admin" \
+    -extensions v3_req -addext "basicConstraints=critical,CA:FALSE" \
+    -noenc -keyout admin-key.pem -out admin-req.pem 2> /dev/null
+
+    # 3b. Use the fake CA to sign the admin CSR, yielding a certificate.
+    openssl x509 -req -CA cacert.pem -CAkey cakey.pem \
+    -copy_extensions copy -days "${DAYS}" \
+    -in admin-req.pem -out admin-cert.pem 2> /dev/null
+    echo "Created admin certificate"
+
+    rm -f node-req.pem admin-req.pem
+
+    ${OLDMASK}
+    popd
+}
+
+setup_security() {
+    # Set up security plugin configuration
```

Comment:
Can you add some comments on what this is doing, e.g. -icl -nhnv 
I also don't understand the -cd; I would have guessed it cds into that directory, but then the cert paths would all be wrong since they wouldn't be under that directory.

Reply:
I'll add a link to the doc I took this from.